project-koku/koku
# [COST-3277] Handle `ValueError` raised by Azure libraries

## Jira Ticket

[COST-3277](https://issues.redhat.com/browse/COST-3277)

## Description

Handle `ValueError` raised by Azure libraries in the same manner we handle `TypeError` and several others exceptions. The specific exception message is logged and a scrubbed message is presented in the UI.

## Testing

1. Checkout Branch
2. Restart Koku
3. Try to add an Azure source with an invalid tenant_id. Any character not in `frozenset("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz" + "0123456789" + "-.")` is invalid.
4. Verify that "Azure client configuration error." is returned to the UI and actual exception message is logged

Or run
```
python koku/manage.py test providers.test.azure.test_provider
```

## Notes

The new Azure libraries raise `ValueError` and not just exceptions from their custom exception classes.
